docs/language_guide.md

```diff
@@ -0,0 +1,619 @@
+# Language Guide
+
+Sysl is a system specification language (and toolkit). You describe applications and data models that make up your system and use `sysl` commands like `sd`, `ints` and `data` to render sequence, integration and data diagrams. Sysl can be extended using plugins. `reljam` is a plugin that reads intermediate representation generated by `sysl` to generate Spring based Java RestController for your API defined in sysl. `gosysl` is another plugin that generates `go` stubs for the same api definition.
+
+Your complete system can be described as forest of trees, where one tree represents one application or data model.
+
+Sysl uses indentation to represent parent child or `has` relationships. E.g. an  `application` has `endpoints` or a `table` has `columns`.
+
+## Concepts
+Sysl allows you to specify [Application](#application) behaviour and [Data Model](#data) that is shared between your applications. Another related concept is of software [Projects](#projects) where you can document what changes happened in each project or a release.
+
+More on this below.
+
+### Application
+An application is an independent entity that provides services via its various `endpoints`. E.g. a `MobileApp` allows users to `Login`, `Search` or `Order` items.
+
+Let's start to define a mobile app with these high level endpoints (or flows):
+```
+MobileApp:
+  Login: ...
+  Search: ...
+  Order: ...
+```
+In the above example, `:` and `...` have special meaning. `:` is used to create a parent-child relationship.
+
+`MobileApp` is a user-defined name that `has` three `endpoints` viz `Login`, `Search` and `Orders`. The way to do that is by having a `newline` and `indent` after `:`.
+
+The `...` (aka shortcut) means, we don't have enough details yet to describe how this endpoint behaves. Sysl allows you to take an iterative approach in documenting the behaviour. You add more as you know more.
+
+### Data
+You will have various kinds of data passing through your systems. Sysl allows you to express ownership, information classification and other attributes of your data in one place.
+
+Continuing with the previous example, let's define a `Server` that expects `LoginData` for the `Login` Flow.
+```
+Server:
+  !type LoginData
+    username <: string
+    password <: string
+```
```

Comment:
not sure if this is the clearest example. in previous example we had a `Login` endpoint / flow. Is `LoginData` of this `Server` related to that, or a different `Login`?